OpenSim/Simulation/Model/PhysicalFrame.cpp

```diff
@@ -237,22 +238,37 @@ void PhysicalFrame::convertDisplayGeometryToGeometryXML(SimTK::Xml::Element& bod
                 geomFile = geomFileIter->getValueAs<SimTK::String>();
             }
             // transform
-            SimTK::Vec6 localXform(0.);
-            SimTK::Xml::element_iterator localXformIter = displayGeomIter->element_begin("transform");
-            if (localXformIter != displayGeomIter->element_end()) {
-                localXform = localXformIter->getValueAs<SimTK::Vec6>();
+            SimTK::Vec6 innerXformVec6(0.);
+            SimTK::Xml::element_iterator innerXformIter = displayGeomIter->element_begin("transform");
+            if (innerXformIter != displayGeomIter->element_end()) {
+                innerXformVec6 = innerXformIter->getValueAs<SimTK::Vec6>();
             }
-            if (localXform.norm() > SimTK::Eps) {
-                // Create a Frame
+            // Old geometry/visibleObject could specify 2 transforms one per DisplayGeometry 
+            // and one for the whole visibleObject.
+            // We'll attach to the frame only if both transforms are identity
+            attachToThisFrame = (innerXformVec6.norm() < SimTK::Eps)
+                && (outerTransformVec6.norm() < SimTK::Eps);
+            // In practice no models were found where both innerTransform and outerTransform were
+            // non trivial or scale factors were specified in both
+            if (!attachToThisFrame) {
+                // Create a Vec6 by composing inner and outer transforms
+                SimTK::Transform outerTransform(outerTransformVec6.getSubVec<3>(3));
+                outerTransform.updR().setRotationToBodyFixedXYZ(outerTransformVec6.getSubVec<3>(0));
+                SimTK::Transform innerTransform(innerXformVec6.getSubVec<3>(3));
+                innerTransform.updR().setRotationToBodyFixedXYZ(innerXformVec6.getSubVec<3>(0));
+                SimTK::Transform composed = outerTransform.compose(innerTransform);
```

Comment:
> where we have no way to verify correctness

I believe we can verify if the composition implemented is what the previous code base was doing quite easily:

1. Create a simple model with just ground in 3.3. Add a `VisibleObject` and assign it a non-simple transform. Inside the `VisibleObject` add a `Mesh` (a box or block) with its own non-simple transform.
2. In 3.3 Visualize (1.) and record its `DecorativeGeometry` transform. This could be copied and pasted into to the test as the `expectedTransform`.
3. Load the model in the current code base that includes the composition.
4. Visualize and extract the `DecorativeGeometry` transform again as the `currentTransform`. 
5. Test that  `currentTransform == expectedTransform`.
6. Visually inspect the block position and orientation between 3.3 and current visualizations